### PR TITLE
ci: Add CloudFormation deployment workflow & update stack template

### PR DESCRIPTION
**Pull request by 433eros**
This PR addresses the following stories:
- PT-186869072: Commit latest revision of LARA CloudFormation stack template (currently used by the most recent staging stack but not production)
- PT-186869461: Add developer friendly workflow for deploying LARA to Learn/NGSA staging & Learn/NGSA production

### CODE REVIEW

**Review by scytacki — APPROVED** [review]
This looks good to me. I didn't review every line carefully though, so waiting for Doug. And also testing it out on staging, and reviewing the logs to make sure it is doing what it should, seems best.

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        run: |-
+          if [[ "${{ github.event.inputs.environment }}" == "" ]]; then
+            test # Fail if it is blank
+          fi
+          echo "Deploying to environment '${{ github.event.inputs.environment }}'"
+      - uses: actions/checkout@v4
+      - name: Select version to deploy or use latest version of project released
+        # Only run this step to lookup the latest pre-release/release version if no 'version' input was specified
+        if: ${{ github.event.inputs.version == '' }}
+        run: |-
```
Comment (dougmartin) [inline .github/workflows/deploy.yml:35]: I'm wondering if we should add a `set -e` at the top of this script.

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            echo "User did not select a release version to deploy."
+            echo -n "Querying API for latest "
+            if [[ "${{ github.event.inputs.environment }}" == *"(Production)"* ]]; then
+              echo -n "release version"
+              latest_github_release_ver=$(gh release view --json 'tagName' --jq '.tagName[1:]' | tr -d '\n')
+            elif [[ "${{ github.event.inputs.environment }}" == *"(Staging)"* ]]; then
+              echo -n "pre-release version"
+              latest_github_release_ver=$(gh release list | grep -oP 'Pre-release\s+v\K\d+\.\d+\.\d-pre\.\d+' | sort -uVr | head -n 1 | tr -d '\n')
+            else
+              echo "Enironment name must include the sub-string \"(Production)\" or \"(Staging)\" to indicate whether it is a staging or production stack"
```
Comment (dougmartin) [inline .github/workflows/deploy.yml:50]: spelling: s/Enironment/Environment/

**Review by dougmartin — APPROVED** [review]
Looks good.  Just had a suggestion to run on exit code checking in the inline shell scripts to make them a little more bullet proof.

**Pull request merged**